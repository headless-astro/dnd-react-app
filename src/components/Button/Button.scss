@import '../../styles/colors';

.button {
  font-weight: 600;
  background-color: $color-secondary;
  padding-top: 20px;
  padding-right: 35px;
  padding-bottom: 22px;
  padding-left: 35px;
  border-radius: 9px;
  white-space: nowrap;
  outline: none;

  &-primary {
    border: none;
    color: $color-white;
  }

  &-primary:hover {
    background-color: $color-secondary-light;
  }

  &-secondary {
    color: $color-secondary;
    background-color: transparent;
    border: 1px solid $color-secondary;
    padding-top: 19px;
    padding-right: 34px;
    padding-bottom: 21px;
    padding-left: 34px;
  }

  &-secondary:hover {
    color: $color-white;
    background-color: $color-secondary;
  }
}
