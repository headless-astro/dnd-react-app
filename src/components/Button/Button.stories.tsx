import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { action } from '@storybook/addon-actions';

import Button, { Props } from './Button';

type StoryProps = Props & {
  text: string;
};

export default {
  component: Button,
  title: '01 - Basic/Button',
  // creates specific argTypes with options
  argTypes: {
    text: {
      defaultValue: 'Hello',
      control: {
        type: 'text',
      },
    },
    type: {
      defaultValue: 'button',
      control: {
        type: 'select',
        options: ['button', 'submit'],
      },
    },
    primary: {
      defaultValue: true,
      control: {
        type: 'boolean',
      },
    },
  },
} as Meta;

export const ButtonStory: Story<StoryProps> = ({ text, primary, type }) => {
  return (
    <Button primary={primary} type={type} onClick={action('clicked')}>
      {text}
    </Button>
  );
};
