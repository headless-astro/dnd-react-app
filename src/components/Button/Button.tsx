import React, { FC } from 'react';

import './Button.scss';

type Props = JSX.IntrinsicElements['button'] & {
  primary?: boolean;
  type: 'button' | 'submit';
};

const Button: FC<Props> = ({ primary, className, children, onClick, type }) => {
  const baseStyle = 'button';
  const buttonStyle = primary ? `button-primary` : `button-secondary`;
  const classNames = className
    ? `${baseStyle}  ${buttonStyle} ${className}`
    : `${baseStyle} ${buttonStyle}`;

  return (
    <button className={classNames} onClick={onClick} type={type === 'button' ? 'button' : 'submit'}>
      {children}
    </button>
  );
};

export default Button;
