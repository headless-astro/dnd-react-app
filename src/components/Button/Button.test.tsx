import React from 'react';
import { shallow } from 'enzyme';

import Button from './Button';

it('renders correctly primary button', () => {
  const wrapper = shallow(
    <Button primary type="button">
      Primary Button
    </Button>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly secondary button', () => {
  const wrapper = shallow(<Button type="button">Secondary Button</Button>);

  expect(wrapper).toMatchSnapshot();
});
