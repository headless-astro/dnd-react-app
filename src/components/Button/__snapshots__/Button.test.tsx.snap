// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly primary button 1`] = `
<button
  className="button button-primary"
  type="button"
>
  <div
    className="button-content"
  >
    Primary Button
  </div>
</button>
`;

exports[`renders correctly secondary button 1`] = `
<button
  className="button button-secondary"
  type="button"
>
  <div
    className="button-content"
  >
    Secondary Button
  </div>
</button>
`;
